### ISSUE TITLE
No 5ghz available for direct wifi pairing

### ISSUE THREAD

**Issue by daven78** [FIRST-Tech-Challenge/FtcRobotController#61]
We had been running our robot controller with version 5 from last season until tonight.  We are in a building that has lots of 2.4ghz wifi connections, so we switched to the 5ghz channel and have been running with no issues.  Since we upgraded to 6.0 robot controller it defaults to a 2.4ghz channel and we can't find any way to connect with 5ghz, its not an option!!!!!  Why? Please help, had to go back to the older robot controller version 5 so that we can use the 5ghz channels.

**Comment by FROGbots-4634** [comment]
What phones are you using?

**Comment by daven78** [comment]
Moto g4's.  We have two rev expansion hubs with the latest firmware.

**Comment by daven78** [comment]
> What phones are you using?
Moto g4's. We have two rev expansion hubs with the latest firmware. We are using blocks if that makes a difference.

**Comment by cmacfarl** [comment]
When you go to the "Change Wifi Direct Channel" activity under "Advanced Settings" do you not see 5Ghz channels?  e.g.  These screenshots are from a Robot Controller running on an E4 running v6.0

![Screenshot_20201211-105505](https://user-images.githubusercontent.com/1435639/101943732-7c3b4f80-3ba0-11eb-8944-19a50f3a2ca0.png)
![Screenshot_20201211-105513](https://user-images.githubusercontent.com/1435639/101943739-7f364000-3ba0-11eb-9942-7ff767a09b7c.png)
![Screenshot_20201211-105527](https://user-images.githubusercontent.com/1435639/101943752-82313080-3ba0-11eb-949a-d609d71fc709.png)

**Comment by daven78** [comment]
No, only the 2.4ghz channels show up.  (1 through 11).  When we ran the older 5.0 version they showed up just like your pictures.  We can't figure out why, I thought maybe something changed in the new app or something.

**Comment by cmacfarl** [comment]
Oops, my bad.  I thought you said E4.

**Comment by NoahAndrews** [comment]
@daven78 Do you have the Moto G4 or the Moto G4 Play? The Moto G4 is an unsupported/not-legal phone, and the Moto G4 Play does not have the hardware for 5 GHz WiFi.

**Comment by daven78** [comment]
Oh, I think it's a G4 Play.  I think we have some E4's also.

**Comment by FROGbots-4634** [comment]
If it's a G4 Play that begs the question of how were you able to use 5GHz before, if it physically does not have a 5GHz radio?

**Comment by cmacfarl** [comment]
Perhaps confusion between G4 and E4 phones?  As I demonstrated earlier today that's pretty easy to do. :-)

**Comment by daven78** [comment]
I will have to check with our programmer.  When I upgraded to 6.0, he may have given me the older phones instead of the ones we were already using.  If that's all it is, that will be easy to fix.  I will check it out and let you guys know.

**Comment by daven78** [comment]
OK, it's the same phones we were using with the 5.? older robot controller.  When we had trouble with the interference of all the wifi in our building we change the channel to the highest one in the 2.4ghz settings :) not the 5ghz band.  That worked well up until we switched to 6.0.  I could have swore it let us use channel 14 on the 2.4 ghz band, but I just tried to repeat that on the g4 plays and could not.  I reinstalled 5,5.1,5.2,5.3,5.4 and 5.5 and could only access channels 1-11.

Anyway, we have a set of E4's and I put 6.0 on them and set them to the top most 5 ghz channel... yahoo!!!! and the interference is minimal, we have pings of 3ms to 20ms.  On the 2.4ghz band we couldn't get it to go below 400ms or just die with "No Heartbeat".  That's not a pretty picture, the robot controller just continuously resets itself.

Thanks for all your input.  I wasn't looking forward to setting up a faraday cage and look like the battle bots on TV. :)
Dave
Team 9978
